packages/native/src/components/WebView.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from "react";
+import { Platform, ScrollView, StyleSheet, ViewStyle } from "react-native";
+import { WebView } from "react-native-webview";
+import {
+  WebViewMessageEvent,
+  WebViewSourceHtml,
+  WebViewSourceUri,
+} from "react-native-webview/lib/WebViewTypes";
+
+import { COMPONENT_TYPES, createSourceProp } from "@draftbit/types";
+
+// Auto-height fix (if this is not present, scrolling on Android does not work)
+const injectFirst = `
+  window.ReactNativeWebView.postMessage(
+    Math.max(document.body.offsetHeight, document.body.scrollHeight)
+  );
+`;
+
+interface Props {
+  source: WebViewSourceUri | WebViewSourceHtml;
+  style?: ViewStyle;
+}
+
+const NativeWebView: React.FC<Props> = ({ source, style }) => {
+  const [height, setHeight] = React.useState(0);
+  const onMessage = (event: WebViewMessageEvent) =>
+    setHeight(Number(event.nativeEvent.data));
+  return (
+    <ScrollView
+      contentContainerStyle={{ flexGrow: 1, height: style?.height || height }}
+    >
+      <WebView
+        source={source}
+        style={style}
+        injectedJavaScript={injectFirst}
+        onMessage={onMessage}
+      />
+    </ScrollView>
+  );
+};
+
+const BrowserWebView: React.FC<Props> = ({ source, style }) =>
+  React.createElement("iframe", {
+    style: StyleSheet.flatten(style),
```

Comment:
do you need to call StyleSheet.flatten here?

Reply:
Yes. The element receives and object for styles, but RN styles are actually numbers representing an object. Flatten returns the actual object, otherwise it will throw an error